From 6525655bbbf9217d434ccf0b34970ac5996916fd Mon Sep 17 00:00:00 2001
From: Ravi B <ravibabu@ti.com>
Date: Mon, 5 Sep 2011 22:57:41 +0530
Subject: [PATCH 0541/1092] musb:cppi41: The epsize register must be multiple
 of 64

In generic rnids mode the epsize register must be
receive length multiple of 64 bytes.

Signed-off-by: Ravi B <ravibabu@ti.com>
Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
---
 drivers/usb/musb/cppi41_dma.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/musb/cppi41_dma.c b/drivers/usb/musb/cppi41_dma.c
index a6b314d..3cf67a2 100644
--- a/drivers/usb/musb/cppi41_dma.c
+++ b/drivers/usb/musb/cppi41_dma.c
@@ -664,6 +664,10 @@ static void cppi41_set_ep_size(struct cppi41_channel *rx_ch, u32 pkt_size)
 	struct cppi41 *cppi = rx_ch->channel.private_data;
 	void *__iomem reg_base = cppi->musb->ctrl_base;
 	u8 ep_num = rx_ch->ch_num + 1;
+	u32 res = pkt_size % 64;
+
+	/* epsize register must be multiple of 64 */
+	pkt_size += res ? (64 - res) : res;
 
 	musb_writel(reg_base, USB_GENERIC_RNDIS_EP_SIZE_REG(ep_num), pkt_size);
 }
-- 
1.7.11.2

